src/client/lib/hooks/useRecaptcha.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export type RenderOptions = {
+  sitekey: string;
+  size?: string;
+  callback: (token: string) => void;
+  'error-callback': (value: undefined) => void;
+  'expired-callback': (value: undefined) => void;
+};
+
+export const recaptchaReady = () =>
+  typeof window !== 'undefined' &&
+  typeof window.grecaptcha !== 'undefined' &&
+  typeof window.grecaptcha.render === 'function';
+
+const useRecaptchaScript = (
+  url = 'https://www.google.com/recaptcha/api.js?render=explicit',
+) => {
+  const [loaded, setLoaded] = React.useState(false);
+
+  React.useEffect(() => {
+    const scriptExists =
+      typeof document !== undefined &&
+      document.getElementById('g-captcha-script');
+
+    if (scriptExists || recaptchaReady()) {
```

Comment:
is it possible for the script to exist before recaptchaReady ?
This might cause problems if the hook is used more than once on a page 🤔

Reply:
good point, I'll have a think about that and get back to you